### PR TITLE
Application setup

### PR DESCRIPTION
**Pull request by joelove**
Add tooling and configure test suite.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,68 @@
...
+
+##### [Postgres](http://www.postgresql.org/)
+
+This project uses a Postgres database by default. You can install Postgres any way you want. On OSX with Homebrew you can:
+
+```
+brew install postgresql
+```
+
+You'll need to create two databases if you're developing locally, one for the application to use and the other for the test suite. You can choose the names of these databases (typically `<app name>` and `<app name>-test`) by setting the environment variables `DATABASE_URL` and `TEST_DATABASE_URL` (see [direnv](#direnv) below). Once you've done that, you can create the databases using the `createdb` command,
```
Comment (maxmckenzie) [inline README.md]: I reckon expanding this step is needed, i'm a newb to Postgres so a link to an intro article would be helpfull.

File: test/factories/sample.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import uuid from 'node-uuid';
+
+export default Factory =>
+  Factory.define('sample')
+    .attr('id', () => uuid.v4());
```
Comment (maxmckenzie) [inline test/factories/sample.js]: is there a cheat sheet for the factory arguments with more info on `uuis.v4()`

**Comment by maxmckenzie** [comment]
Looks good to me.

File: README.md (hunk 1 above)
Comment (joelove) [inline README.md]: Good idea.

File: test/factories/sample.js (hunk 2 above)
Comment (joelove) [inline test/factories/sample.js]: `uuid.v4()` is just a method on the `node-uuid` module (see line 1). It's an NPM package that generates UUIDs for you. For the factories (which use Rosie, another NPM package) you can just any values you want in. For example, you might want a plain string, a `true` value, or to generate a random string using another package.

In terms of data types in the database, these are predominantly defined by Knex (which is the query builder and handles things like migrations and adding seed data) and generally mirror types available in Postgres databases (SQL+). I'll add a link to the Knex documentation somewhere below.

Also, there are model data types to consider. Whilst these loosely mirror the same types available in Knex (and by extension the database itself) there are some notable differences. I'll add a link to the Bookshelf docs too.

**Comment by joelove** [comment]
:shipit:

**Pull request merged**